app/vmbackup/snapshot/snapshot.go

```diff
@@ -0,0 +1,85 @@
+package snapshot
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+)
+
+type snapshot struct {
+	Status   string `json:"status"`
+	Snapshot string `json:"snapshot"`
+	Msg      string `json:"msg"`
+}
+
+// TakeSnapshot creates a snapshot and the provided api endpoint and returns
+// the snapshot name
+func TakeSnapshot(endpoint string) (string, error) {
+	u, err := url.Parse(endpoint)
+	if err != nil {
+		return "", err
+	}
+	u.Path = path.Join(u.Path, "/snapshot/create")
```

Comment:
I think that it would be better if `-snapshotURL` contains the full url with `/snapshot/create` in the end, like the following: `http://victoriametrics:8428/snaphsot/create`. An example url should be put in `-snapshotURL` description. This will make it possible to provide alternative url path, which could be rewritten by a proxy. Additionally to this the path could contain auth tokens such as `-snapshotAuthKey` defined in single-node VictoriaMetrics. See https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/README.md#security for details.

Reply:
We still need to switch between /create and /delete so it would be best to stop at http://victoriametrics:8428/snaphsot

Reply:
But this won't allow passing optional `authKey` query arg as described at https://github.com/VictoriaMetrics/VictoriaMetrics/blob/master/README.md#security :(

How about defining two flags: `-snapshot.createURL` and `-snapshot.deleteURL`? If `-snapshot.deleteURL` isn't set, then it could be derived from `-snasphot.createURL` by substituting `/create` with `/delete` in it. This should be documented in flags' description.

Reply:
Thanks for the input, does this line up with what you were thinking @valyala ?